subsys/bluetooth/mesh/settings.c

```diff
@@ -2424,22 +2436,27 @@ void bt_mesh_clear_cdb_app_key(struct bt_mesh_cdb_app_key *key)
 }
 
 int bt_mesh_model_data_store(struct bt_mesh_model *mod, bool vnd,
-			     const void *data, size_t data_len)
+			const char *name, const void *data,
+			size_t data_len)
 {
-	char path[20];
+	u16_t mod_key;
+	char path[30];
 	int err;
 
-	encode_mod_path(mod, vnd, "data", path, sizeof(path));
+	mod_key = (((u16_t)mod->elem_idx << 8) | mod->mod_idx);
+
+	if (name) {
+		snprintk(path, sizeof(path), "bt/mesh/%s/%x/%s/%s",
+			vnd ? "v" : "s", mod_key, "data", name);
+	} else {
+		snprintk(path, sizeof(path), "bt/mesh/%s/%x/%s",
+			vnd ? "v" : "s", mod_key, "data");
+	}
```

Comment:
Can go back to old code now to simplify this:

```C
encode_mod_path(mod, vnd, "data", path, sizeof(path));
if (name) {
    strcat(path, "/");
    strncat(path, name, 8);
}
```

Reply:
You're too smart